lib/pbench/server/__init__.py

```diff
@@ -125,6 +117,72 @@ def mocked_time():
             + " ".join([f"WONT-INDEX.{i:d}" for i in range(1, 12)])
         )
 
+    def _get_conf(self, section, option):
+        """
+        _get_conf - get the option from the section, raising
+        BadConfig Error if it is empty, return the option value as a string.
+        """
+        try:
+            option_val = self.conf.get(section, option)
+        except (NoOptionError, NoSectionError) as exc:
+            raise BadConfig(str(exc))
+        else:
+            if not option_val:
+                raise BadConfig(f"option {option} in section {section} is empty")
+
+        return option_val
+
+    def get_conf(self, name, section, option, logger):
+        """
+        get_conf - get the option from the section, raising BadConfig Error
+        if it is empty, return the option value as a string.
+        """
+        try:
+            option_val = self._get_conf(section, option)
+        except BadConfig as exc:
+            logger.error("Bad {}= '({})'", name, exc)
+            return None
+
+        return option_val
+
+    def _get_valid_dir_option(self, req_val, section, option):
+        """_get_valid_dir_option - get the directory option from the
+        given section, raising BadConfig Error if the path is not resolved
+        or is not a directory, returning a Path directory object.
+        """
+        dir_val = self._get_conf(section, option)
+        dir_path = self.get_valid_dir_option(req_val, dir_val, None)
+        if not dir_path:
+            raise BadConfig(f"Bad {req_val}={dir_val}")
+
+        return dir_path
+
+    def get_valid_dir_option(self, req_val, dir_val, logger):
+        """get_valid_dir_option - get the directory path raising an Error
+        if it is not a directory, returning a Path directory object.
+        """
+        try:
+            dir_path = Path(dir_val).resolve(strict=True)
+        except FileNotFoundError:
+            if logger:
+                logger.error(
+                    "The {} directory, '{}', does not resolve to a real location",
+                    req_val,
+                    dir_val,
+                )
+            return None
+        else:
+            if not dir_path.is_dir():
+                if logger:
+                    logger.error(
+                        "The ARCHIVE directory, '{}', does not resolve to a directory ('{}')",
```

Comment:
This should not refer to ARCHIVE: it should be a generic {} filled in from an argument.